Sprint 08

Azonaz opened the pull request.

Azonaz committed:
Закончена задача по 8 спринту:
- создан новый проект,
- сверстан Launch Screen,
- сверстан главный экран с заданными параметрами.

Azonaz committed:
поправила ошибку

Helvetii951 commented on ImageFeed/ImagesList/ImagesListCell.swift:
```diff
@@ -0,0 +1,27 @@
+        cellImage.layer.insertSublayer(gradientLayer, at: 0)
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        gradientLayer.frame = CGRect(x: 0, y: cellImage.bounds.height - 30, width: cellImage.bounds.width, height: 30)
+    }
```
За знание о `layoutSubviews()` мое уважение🤝

Helvetii951 commented on ImageFeed/ImagesList/ImagesListCell.swift:
```diff
@@ -0,0 +1,27 @@
+
+final class ImagesListCell: UITableViewCell {
+    static var reuseIdentifier = "ImagesListCell"
+    var gradientLayer: CAGradientLayer!
```
`gradientLayer` можно сделать private. 

Рекомендую сразу делать переменные и функции с минимальной областью видимости. По мере необходимости ее менять.

Helvetii951 commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,75 @@
+        let isLiked = indexPath.row % 2 == 0
+        let likedImage = isLiked ? UIImage(named: "LikeOn") : UIImage(named: "LikeOff")
+        cell.likeButton.setImage(likedImage, for: .normal)
+        let selectView = UIView()
+        selectView.backgroundColor = UIColor.darkGray.withAlphaComponent(0.5)
+        cell.selectedBackgroundView = selectView
```
Выделение ячейки по проекту не предполагается. Лучше убрать.

Helvetii951 commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -0,0 +1,75 @@
+
+extension ImagesListViewController {
+    
+    func configCell(for cell:ImagesListCell, with indexPath: IndexPath) {
+        guard let image = UIImage(named: photosName[indexPath.row]) else {
+            return
+        }
+        cell.cellImage.image = image
+        cell.dateLabel.text = dateFormatter.string(from: Date())
```
Тоже можно сделать функцию приватной.

Azonaz committed:
Внесены исправления по замечаниям ревьюера:
- исправлен размер кнопки Лайк,
- исправления приватности,
- убран лишний код =)
